File: ytrue-bean/src/main/java/com/ytrue/bean/dataobject/system/SysDept.java
package com.ytrue.bean.dataobject.system;

import com.baomidou.mybatisplus.annotation.TableName;
import com.ytrue.bean.dataobject.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import java.io.Serial;

/**
 * @author ytrue
 * @date 2022-08-04
 * @description 部门实体类
 */
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@Data
@TableName("sys_dept")
public class SysDept extends BaseEntity {

    @Serial
    private static final long serialVersionUID = 2021887051210043344L;


    @Schema(description = "上级部门")
    private Long pid;

    @Schema(description = "子节点数目", hidden = true)
    private Integer subCount;

    @Schema(description = "名称")
    private String deptName;

    @Schema(description = "负责人")
    private String leader;

    @Schema(description = "联系电话")
    private String phone;

    @Email
    @Schema(description = "邮箱")
    private String email;

    @NotNull
    @Schema(description = "是否启用")
    private Boolean status;

    @Schema(description = "排序")
    private Integer deptSort;


}

File: ytrue-admin-server-csharp/Ytrue.Models/Entities/System/SysDept.cs
using System.ComponentModel.DataAnnotations.Schema;
using Ytrue.Models.Bases;

namespace Ytrue.Models.Entities.System;

/// <summary>
/// 系统部门实体
/// </summary>
[Table("sys_dept")]
public class SysDept : BaseEntity
{
    /// <summary>
    /// 上级部门
    /// </summary>
    [Column("pid")]
    public long Pid { get; set; }

    /// <summary>
    /// 子部门数目
    /// </summary>
    [Column("sub_count")]
    public int SubCount { get; set; }

    /// <summary>
    /// 名称
    /// </summary>
    [Column("dept_name")]
    public string DeptName { get; set; }

    /// <summary>
    /// 排序
    /// </summary>
    [Column("dept_sort")]
    public int DeptSort { get; set; }

    /// <summary>
    /// 负责人
    /// </summary>
    [Column("leader")]
    public string Leader { get; set; }

    /// <summary>
    /// 邮箱
    /// </summary>
    [Column("email")]
    public string Email { get; set; }

    /// <summary>
    /// 联系电话
    /// </summary>
    [Column("phone")]
    public string Phone { get; set; }

    /// <summary>
    /// 状态:1=启用,0=禁用
    /// </summary>
    [Column("status")]
    public bool Status { get; set; }
}